packages/superset-ui-chart/src/components/SuperChart.tsx

```diff
@@ -110,35 +111,48 @@ export default class SuperChart extends React.PureComponent<Props, {}> {
       FallbackComponent,
       onErrorBoundary,
       Wrapper,
+      queryData,
       ...rest
     } = this.props as PropsWithDefault;
 
-    const chartWithoutWrapper = (
-      <SuperChartCore
-        ref={this.setRef}
-        id={id}
-        className={className}
-        chartType={chartType}
-        chartProps={this.createChartProps({
-          ...rest,
-          height,
-          width,
-        })}
-        preTransformProps={preTransformProps}
-        overrideTransformProps={overrideTransformProps}
-        postTransformProps={postTransformProps}
-        onRenderSuccess={onRenderSuccess}
-        onRenderFailure={onRenderFailure}
-      />
-    );
-    const chart = Wrapper ? (
-      <Wrapper width={width} height={height}>
-        {chartWithoutWrapper}
-      </Wrapper>
-    ) : (
-      chartWithoutWrapper
-    );
-
+    const chartProps = this.createChartProps({
+      ...rest,
+      queryData,
+      height,
+      width,
+    });
+
+    let chart;
+    // Render the no results component if the query data is null or empty
+    if (
+      queryData == null ||
+      queryData.data === null ||
+      (Array.isArray(queryData.data) && queryData.data.length === 0)
+    ) {
+      chart = <NoResultsComponent height={height} width={width} />;
```

Comment:
Perhaps it should take `id` and `className` as well?

Reply:
makes sense, added